File: twitterClient/twitterClient/src/stomp/frames/ConnectFrame.cpp
//
//  ConnectFrame.cpp
//  Object to represent the CONNECT frame type
//
//  Authors:    Ken Saggy
//              Lidan Hifi
//
//  Copyright 2013 Lidan Hifi, Ken Saggy

#include "../../../include/stomp/frames/ConnectFrame.h"

namespace stomp {
    ConnectFrame::ConnectFrame(): StompFrame(CONNECT_FRAME_NAME) {
        addHeaderPair("accept-version", "1.2");
    }
    
    ConnectFrame::ConnectFrame(std::string host, std::string login, std::string password): StompFrame(CONNECT_FRAME_NAME) {
        addHeaderPair("accept-version", "1.2");
        addHeaderPair("host", host);
        addHeaderPair("login", login);
        addHeaderPair("passcode", password);
    }
    
    void ConnectFrame::accept(FrameVisitor *visitor)
    {
    }
}

File: twitterEngine/src/spl/stomp/protocol/ConnectFrame.java
package spl.stomp.protocol;

/**
 * @name Connect Frame
 * @author  Lidan Hifi
 * @author  Ken Saggy
 * 
 * CONNECT stomp frame
 */
class ConnectFrame extends StompFrame {
    public ConnectFrame() {
        super(StompCommand.CONNECT);
    }
    
    public ConnectFrame(String host, String login, String passcode) {
        super(StompCommand.CONNECT);
        addFrameHeader("accept-version", "1.2");
        addFrameHeader("host", host);
        addFrameHeader("login", login);
        addFrameHeader("passcode", passcode);
    }
    
    @Override
    protected String[] getMandatoryHeaders() {
        return new String[] {"accept-version", "host", "login", "passcode"};
    }
}
